pkg/registry/certificates/certificates/strategy.go

```diff
@@ -235,6 +270,23 @@ type csrApprovalStrategy struct {
 
 var ApprovalStrategy = csrApprovalStrategy{Strategy}
 
+// GetResetFields returns the set of fields that get reset by the strategy
+// and should not be modified by the user.
+func (csrApprovalStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
+		"certificates.k8s.io/v1": fieldpath.NewSet(
+			fieldpath.MakePathOrDie("spec"),
+			fieldpath.MakePathOrDie("status", "certificate"),
```

Comment:
This rule is not explicit enough (but it does the job for now).
Technically we would need a path that says "reset everything in this path, but conditions"
As a csr status only has conditions and the cert, we should be fine, but i'm not completely satisfied.